_posts/2017-09-05-Rust-2017-Survey-Results.md

```diff
@@ -0,0 +1,126 @@
+
+![Chart: 17.5% daily, 43.3% weekly, 24.4% monthly, 14.9% rarely](/images/2017-09-05-Rust-2017-Survey-Results/how_often_use_rust.png "17.5% daily, 43.3% weekly, 24.4% monthly, 14.9% rarely")
+
+Despite the rising amount of code developers are working with, we're seeing a small downtick in both daily and weekly users.  Daily users have fallen from 19% to 17.5%, and weekly users have fallen from 48.8% to 43.3%. This could be a natural transition in this stage of our growth, as a broader range of developers begin using Rust.
+
+# Path to Stability
+
+![Chart: 92.5% no, 7.5% yes](/images/2017-09-05-Rust-2017-Survey-Results/upgrade_stable.png "92.5% no, 7.5% yes")
+
+In the last year, we made big strides in breakages caused by releases of the compiler.  Last year, 16.2% respondents said that upgrading to a new stable Rust compiler broke their code.  This year, that number has **fallen to 7.5% of respondents**.  This is a huge improvement, and one we're proud of, though we'll continue working to push this down even further.
+
+![Chart show strong support for nightly and current stable releases](/images/2017-09-05-Rust-2017-Survey-Results/which_version.png)
+
+Developers have largely opted to move to nightly or a recent stable (with some on beta), showing that developers are eager to upgrade and do so quickly.  This simplifies the support structure a bit from last year, where developers were on a wider range of versions.
+Unfortunately, despite our efforts with procedural macros and helping move crates like Serde to stable, we still have work to do to promote people moving away from the nightly Rust compiler.  This year shows an increase in nightly users, now at 1,852 votes it represents 51.6% of respondents using nightly, up from 48.8% of last year.
+
+# How we use Rust
+
+![Chart: 90.2% rustup, 18.9% linux distros, 5% homebrew, 4.7% official .msi, 3.1% official tarball, 1.4% official mac pkg](/images/2017-09-05-Rust-2017-Survey-Results/rustup.png "90.2% rustup, 18.9% linux distros, 5% homebrew, 4.7% official .msi, 3.1% official tarball, 1.4% official mac pkg")
+*(hover for more info)*
+
+One of the big success stories with Rust tooling was rustup, the Rust toolchain installer.  Last year, we saw a wide diversity in ways people were installing Rust.  This year, many of these have moved to using rustup as their main way of installing Rust, totalling now 3,205 of the responses, which moves it from last year's 52.8% to **90.2%**.
+
+![Chart: 80.9% Linux, 35.5% macOS, 31.5% Windows, 3.2% BSD-variant](/images/2017-09-05-Rust-2017-Survey-Results/platforms_on.png)
+
+Linux still features prominently as one of the main platforms Rust developers choose.  Of note, we also saw a rise in the use of Windows as a developer platform at 1,130 of the 3,588 total respondents, putting it at **31.5% of respondents**, up from 27.6% of last year.
```

Comment:
A 3% rise in the number of Windows users doesn't seem very note-worthy to me - how does it compare to movement on the other platforms? Is there a reason to call out Windows in particular?

Reply:
It's 4%, and it's the only change since last year, so I called it out :)